docs/source/_static/custom.css

```
@@ -0,0 +1,3 @@
+.section dl.py:not(:first-of-type) {
```

Achieve the same effect without the `:not(...)`  stuff that might not be supported everywhere.

```suggestion
.section dl.py + dl.py {
```

---

If you're worried about the `:not()` selector, I don't think it's worth worrying about with 95% support and working in every modern browser under the sun: https://caniuse.com/#feat=mdn-css_selectors_not

The `:first-of-type` selector is almost as broadly supported: https://caniuse.com/#feat=mdn-css_selectors_first-of-type

Only depressingly old browsers will have trouble with this QoL style.

That said, if you have another suggestion for avoiding extra space between the dl and headings, I'm all eyes. This was just the first solution I devised. It might have other edge cases I didn't consider.

---

I checked how `:first-of-type` works exactly, because I wasn't sure, and it's slightly different from `x + y`:

```
<div>
    <h2></h2>
    <dl></dl> <!-- :first-of-type -->
    <dl></dl> <!-- :not(:first-of-type) ; dl + dl--> 
    <h2></h2>
    <dl></dl><!-- :not(:first-of-type) -->
    <dl></dl><!-- :not(:first-of-type) ; dl + dl -->
</div>
```

In our case, I think Sphinx never include other elements than `dl` at the same level, so it's technically the same in our context. However, in the general case, they don't achieve the same thing.

It's up to you. I prefer the `x + y` approach because I works better for me and how I represent things.

---

Looking back at the above metrics, plus https://caniuse.com/mdn-css_selectors_adjacent_sibling: I finally choose to stick with how I originally wrote the selector. Real-world stats say it has slightly _higher_ support than `+` (I was surprised too), and it's safer in case we end up doing something later that makes Sphinx give us an unexpected `dl` at the same level as something else.

---

It's a trick of the light.